vim-plugin/plugin/ethersync.lua

```diff
@@ -106,38 +113,37 @@ local function on_buffer_open()
         connect()
     end
 
-    files[filename] = {
-        -- Number of operations the daemon has made.
-        daemon_revision = 0,
-        -- Number of operations we have made.
-        editor_revision = 0,
-    }
-
     local uri = "file://" .. filename
-    -- TODO: If open fails, abort adding it to the changetracker (issue #88).
-    send_request("open", { uri = uri })
+    send_request("open", { uri = uri }, function()
+        files[filename] = {
+            -- Number of operations the daemon has made.
+            daemon_revision = 0,
+            -- Number of operations we have made.
+            editor_revision = 0,
+        }
 
-    -- Vim enables eol for an empty file, but we do use this option values
-    -- assuming there's a trailing newline iff eol is true.
-    if vim.fn.getfsize(vim.api.nvim_buf_get_name(0)) == 0 then
-        vim.bo.eol = false
-    end
+        -- Vim enables eol for an empty file, but we do use this option values
+        -- assuming there's a trailing newline iff eol is true.
+        if vim.fn.getfsize(vim.api.nvim_buf_get_name(0)) == 0 then
+            vim.bo.eol = false
+        end
 
-    changetracker.track_changes(0, function(delta)
-        files[filename].editor_revision = files[filename].editor_revision + 1
+        changetracker.track_changes(0, function(delta)
+            files[filename].editor_revision = files[filename].editor_revision + 1
 
-        local rev_delta = {
-            delta = delta,
-            revision = files[filename].daemon_revision,
-        }
+            local rev_delta = {
+                delta = delta,
+                revision = files[filename].daemon_revision,
+            }
 
-        local params = { uri = uri, delta = rev_delta }
+            local params = { uri = uri, delta = rev_delta }
 
-        send_request("edit", params)
-    end)
-    cursor.track_cursor(0, function(ranges)
-        local params = { uri = uri, ranges = ranges }
-        send_notification("cursor", params)
+            send_request("edit", params)
+        end)
+        cursor.track_cursor(0, function(ranges)
+            local params = { uri = uri, ranges = ranges }
+            send_request("cursor", params)
```

Comment:
For the time being, I thought it might be interesting to see if our "cursor" messages fail, as well (for example, because they're being sent for ignored files).

Reply:
makes sense :) I added a comment to it.